lib/logstash/outputs/gelf.rb

```diff
@@ -81,12 +81,22 @@ class LogStash::Outputs::Gelf < LogStash::Outputs::Base
   config :short_message, :validate => :string, :default => "short_message"
 
   public
+
+  def inject_client(gelf)
+    @gelf = gelf
+    self
+  end
+
+  def gelf
+    @gelf
+  end
+
   def register
     require "gelf" # rubygem 'gelf'
     option_hash = Hash.new
 
     #@gelf = GELF::Notifier.new(@host, @port, @chunksize, option_hash)
-    @gelf = GELF::Notifier.new(@host, @port, @chunksize)
+    @gelf ||= GELF::Notifier.new(@host, @port, @chunksize)
```

Comment:
It's not necessary to create an `inject_client` method in that context, what is evil is our usage or `ivar` in the plugins.
You create a reader method specifically for the gelf client.

```ruby
def gelf
  @gelf ||= if the bundle of jars is done in the `rake publish` it should work out of the box with jarvis 
end
```
And in the test you can mock it.

```
output = Logstash::Output::Gelf.new
expect(output).to receive(:client).and_return(mocked_gelf)
```